OTA fixes (and more)

puddly opened the pull request:
This PR tracks minor fixes I had to make to get things running.

Corresponding Core branch: https://github.com/dmulcahey/home-assistant/compare/dm/zha-external-zha...puddly:core:puddly/zha-external-zha

puddly committed:
`golbal` -> `global`

puddly committed:
Allow providing event name as context

puddly committed:
Offload on/off state computation to Core

puddly committed:
Use a separate event for `LevelChangeEvent`

puddly committed:
Do not set OTA entity state now that we do not compute it

puddly committed:
Stop OTA when we fail

puddly commented:
Some entity names are a little messed up:

<img width="826" alt="image" src="https://github.com/zigpy/zha/assets/32534428/de5b707a-64d2-4768-b0d0-078e8333fe70">

puddly committed:
Set up public accessors for `name` and `translation_key`

puddly committed:
Add icon and entity category as well

puddly committed:
Add a translation key to the `identify` button

TheJulianJES commented on zha/application/platforms/button/__init__.py:
```diff
@@ -132,6 +132,7 @@ def create_platform_entity(
 
     _attr_device_class = ButtonDeviceClass.IDENTIFY
     _attr_entity_category = EntityCategory.DIAGNOSTIC
+    _attr_translation_key = "identify"
```
I don't think this should have a translation key for HA. It already has the "identify" device class which is enough for the name. Same as battery entities (only use device class, not another explicit "Battery" name/translation).

puddly replied:
You're right, this can be removed without issue.

I think originally I was setting `_attr_name = None`, which again defaulted to the device name. Fixed by not setting the attribute unless it has a value.

puddly committed:
Revert "Add a translation key to the `identify` button"

This reverts commit 274530612db975cb38dc4bcd418ee40e3146200b.

puddly committed:
Drop all simple icons, HA already uses icon translations

puddly committed:
Drop `icon` entirely

puddly committed:
Clarify `type: ignore`s
The `if self.is_transitioning` check at the top of the function "pins"
the value during the execution context in mypy's eyes. There doesn't
seem to be a way to force mypy to consider the property "`volatile`".

puddly committed:
Add public accessors for device and state class

puddly committed:
Use parentheses for clarity

puddly committed:
Consolidate `get_entity` into `tests.common`

puddly committed:
Unwind stack frames to log correctly in ZCL handlers

puddly committed:
Fix cover level change event

puddly committed:
Bump ruff and run `ruff-format` during pre-commit

puddly committed:
Log when emitting a ZHA event

puddly committed:
Re-add `--fix`

puddly committed:
Fix most unit tests

puddly committed:
Fix `update` platform tests

puddly committed:
Fix entity ID in counter unit test

puddly committed:
Consolidate entity information into `info_object`

puddly committed:
Rename `name` to `fallback_name` to clarify intent
The name should be localized

puddly committed:
Drop `internal_name` as well

puddly committed:
Attach `unique_id` to `Device`

puddly committed:
Compute the final `entity_id` only in `BaseEntity`

puddly committed:
Fix unit tests

puddly committed:
Include the device unique ID in the counter unique ID

puddly committed:
Ignore `_attr_translation_key` on counters

puddly committed:
Expose `entity_registry_enabled_default`

puddly committed:
Correctly construct counter unique IDs

puddly committed:
Mark `entity_id` for removal

puddly committed:
Get rid of `entity_id`

puddly committed:
Handle some TODOs

puddly committed:
Update dependencies

puddly committed:
Drop unused dependencies
